enhancements/hypershift/kubevirt-static-ip-address-management.md

```diff
@@ -0,0 +1,531 @@
+
+## Motivation
+
+Currently, HyperShift with the KubeVirt provider relies on DHCP for IP address allocation to guest cluster nodes. This approach introduces a DHCP dependency and for some customers, DHCP is not desired for various reasons, like latency, complexity or reliability.
+
+By implementing static IPAM for KubeVirt-based HyperShift clusters, we enable deployment in a broader range of environments and provide operators with greater control over network configuration.
+
+### User Stories
+
+* As a cluster administrator I want to use Ingress Controller(s) on nodes with predefined IPs, so that I can update their upstream DNS and LB beforehand.
+* As a cluster administrator, I need to be able to configure static IPs for my tenant workers, since DHCP server is not available in my network.
+
+### Goals
+
+- Enable static IP address assignment for HyperShift guest cluster nodes running on KubeVirt
+- Provide a flexible API for defining IP address pools with support for various IP range formats (CIDR, hyphenated ranges, single IPs)
+- Support both IPv4 and IPv6 addressing
+- Supporting dynamic IP pool expansion or runtime IP reallocation
+- Support KubeVirt VMs live migration so the assigned network configuration stick to the VMs after live migrating them.
+- Recover pools from cluster api provider kubevirt pod restarts or node restarts.
+
+### Non-Goals
+
+- Replacing existing network configuration mechanisms in standalone OpenShift clusters
+- Implementing a full-featured IPAM solution with conflict detection across multiple management clusters
+- Providing network configuration for components outside of HyperShift guest cluster nodes
+- Support complex network configurations like bonding of interfaces and others usually configured with ignition
+
+## Proposal
+
+This proposal introduces static IPAM capabilities through modifications to three key components:
+
+1. **Cluster API Provider KubeVirt**: Extend the API to support IP pool definitions and network configuration, implement IP allocation logic from defined pools
+2. **HyperShift**: Add network configuration options to the KubeVirt platform specification at nodepool, implement translation from this network configuration to capk ip pool and openstack network config
+3. **CoreOS Afterburn**: Enable parsing and application of OpenStack network config standard data as dracut network kernel args from cloud-init config drive for KubeVirt VMs, this is similar what is done at proxmoxve provider.
+
+### Workflow Description
+
+**cluster administrator** is a human user responsible for deploying a HyperShift hosted cluster.
+
+1. Disable the default pod network and select the network to replace it:
+   - Sets `AttachDefaultNetwork=false` in the NodePool specification
+   - Specifies an additional network attachment that will replace the default network
+   - Ensures the static IP configuration matches the custom network's name
```

Comment:
I read the words, but I have no clue on what "form" this will be implemented as. 

Maybe an example would be good to have.